File: VB/EUD_RenameToolboxItems/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports System.Drawing.Design
Imports DevExpress.XtraReports.UserDesigner
' ...

Namespace EUD_RenameToolboxItems
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
	' Create a new End-User Designer.
	Dim designForm As New XRDesignForm()

	' Handle the DesignPanelLoaded event.
	AddHandler designForm.DesignMdiController.DesignPanelLoaded, AddressOf DesignMdiController_DesignPanelLoaded

	' Load the report into the designer. 
	designForm.OpenReport(New XtraReport1())

	' Show the End-User Designer window.
	designForm.Show()
End Sub

Private Sub DesignMdiController_DesignPanelLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
	' Get the toolbox service.
	Dim ts As IToolboxService = CType(e.DesignerHost.GetService(GetType(IToolboxService)), IToolboxService)

	' Get a collection of toolbox items.
	Dim coll As ToolboxItemCollection = ts.GetToolboxItems()

	' Iterate through toolbox items.
	For Each item As ToolboxItem In coll
		' Add the "Cool" prefix to all toolbox item names.
		item.DisplayName = "Cool " & item.DisplayName
	Next item
End Sub

	End Class
End Namespace

File: CS/EUD_RenameToolboxItems/Form1.cs
using System;
using System.Windows.Forms;
using System.Drawing.Design;
using DevExpress.XtraReports.UserDesigner;
// ...

namespace EUD_RenameToolboxItems {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            // Create a new End-User Designer.
            XRDesignForm designForm = new XRDesignForm();

            // Handle the DesignPanelLoaded event.
            designForm.DesignMdiController.DesignPanelLoaded +=
                new DesignerLoadedEventHandler(DesignMdiController_DesignPanelLoaded);

            // Load the report into the designer. 
            designForm.OpenReport(new XtraReport1());

            // Show the End-User Designer window.
            designForm.Show();
        }

        private void DesignMdiController_DesignPanelLoaded(object sender, DesignerLoadedEventArgs e) {
            // Get the toolbox service.
            IToolboxService ts = (IToolboxService)e.DesignerHost.GetService(typeof(IToolboxService));

            // Get a collection of toolbox items.
            ToolboxItemCollection coll = ts.GetToolboxItems();

            // Iterate through toolbox items.
            foreach (ToolboxItem item in coll) {
                // Add the "Cool" prefix to all toolbox item names.
                item.DisplayName = "Cool " + item.DisplayName;
            }
        }

    }
}
